### PR TITLE
Simplify logic of defunct trigger removal

### PR DESCRIPTION
**Pull request by eric-wieser**
I'm not sure if this is actually faster, but it does reduce the big-O complexity a little.

More usefully, it makes the code easier to reason about by never visiting the same trigger more than once, Previously, this would cause it to repeatedly insert then immediately delete empty lists into `self._trigger2coros`, once per triggered coroutine

### CODE REVIEW

File: cocotb/scheduler.py [hunk 1]
```diff
@@ -392,19 +392,24 @@ def react(self, trigger, depth=0):
...
+            t
+            for coro in scheduling
+            for t in self._coro2triggers[coro]
+        } - {trigger}
+
+        for pending in other_triggers:
+            # every coroutine waiting on this trigger is already being woken
+            if scheduling_set.issuperset(self._trigger2coros[pending]):
+                if pending.primed:
+                    pending.unprime()
```
Comment (eric-wieser) [inline cocotb/scheduler.py]: I suspect that `pending.primed` is now always true here.

**Comment by eric-wieser** [comment]
[5% faster](https://travis-ci.org/potentialventures/cocotb/jobs/476209976#L8861) (3.02s) than [an earlier build](https://travis-ci.org/potentialventures/cocotb/jobs/475560168#L8993) (3.19s), but I doubt that the measurements are very precise.

**Comment by imphil** [comment]
I like it!

File: cocotb/scheduler.py (hunk 1 above)
Comment (imphil) [reply cocotb/scheduler.py]: I didn't look super closely, but that statement seems to be true. But it shouldn't harm either.

**Pull request merged**